src/00b3_survey_mics/script.R

```diff
@@ -211,146 +290,183 @@ mics_final <- circ_recoded %>%
 if (n > 0 && n < 20) {
- mics_final <- mics_final %>% 
-   filter(is.na(circ_age) | is.na(age) | circ_age <= age)
+  mics_final <- mics_final %>%
+    filter(is.na(circ_age) | is.na(age) | circ_age <= age)
+} else {
+  stop("Too many errors in circ_age > age; something likely wrong")
 }
 
 # add area_id to dataset
-areas_join <- areas %>% 
+areas_join <- areas %>%
   # # hh7 (i.e. area_name) maps to area lev 1, filter to avoid duplication
   filter(
-    (area_level == 1) |
-    # special cases
-    (iso3 == "MWI" & area_level %in% c(3, 5)) |  # need lev 3 for Mzimba
-    (iso3 == "NGA" & area_level == 2)
-  ) %>% 
-  select(area_id, area_name, area_level) %>% 
-  arrange() 
+    (area_level == 1 & !iso3 %in% c("MWI", "NGA")) |
+      # special cases
+      (iso3 == "MWI" & (area_name == "Mzimba" & area_level == 3 | area_level == 5)) |  # need lev 3 for Mzimba
+      (iso3 == "NGA" & area_level == 2)
+  ) %>%
+  select(iso3, area_id, area_name, area_level) %>%
+  arrange()
 
 # remove possible duplicate, lower level areas from areas_join (for MWI)
-duplicate_areas <- areas_join %>% 
-  group_by(area_name) %>% 
+duplicate_areas <- areas_join %>%
+  group_by(iso3, area_name) %>%
   summarise(
-    occurences = n(), 
-    area_id = area_id, 
+    occurences = n(),
+    iso3 = iso3,
+    area_id = area_id,
     area_level = area_level,
     max_area_level = max(area_level),
     .groups = "drop"
-  ) %>% 
-  filter(occurences > 1, area_level != max_area_level) %>% 
-  select(area_name, area_id)
-
-areas_join <- areas_join %>% 
-  anti_join(duplicate_areas, by = c("area_id", "area_name")) %>% 
-  select(area_id, area_name)
-  
-# split by iso3 before joining with areas_join, again to avoid duplication 
-mics_final_list <- split(mics_final, mics_final$iso3)
-mics_final <- bind_rows(lapply(seq_along(mics_final_list), function(i) {
-  mics_final_list[[i]] %>% 
-    left_join(
-      filter(areas_join, grepl(names(mics_final_list)[i], area_id)), 
-      by = "area_name"
-    )
-}))
+  ) %>%
+  filter(occurences > 1, area_level != max_area_level) %>%
+  select(iso3, area_name, area_id)
 
-# check that no duplication has occurred (seems to be duplication for MWI!)
-mics_final %>% 
-  mutate(
-    check = paste(
-      survey_id, individual_id, cluster_id, household, line, circ_status, 
-      circ_age, circ_where, area_name, iso3, sex, age
-    )
-  ) %>% 
-  filter(duplicated(check)) %>% 
-  nrow() == 0
+areas_join <- areas_join %>%
+  anti_join(duplicate_areas, by = c("iso3", "area_id", "area_name")) %>%
+  select(iso3, area_id, area_name)
 
-# finally, assign any surveys with NA for area_id to the national level
-mics_final <- mics_final %>% 
-  mutate(
-    across(contains("area"), ~ ifelse(is.na(area_id), iso3, .)),
-    area_name = ifelse(
-      area_name == iso3, 
-      countrycode::countrycode(
-        area_name, origin = "iso3c", destination = "country.name"
-      ), 
-      area_name
-    )  
-)
```

Comment:
I didn't have any cases of this when I finished, so I removed. But in general, I would be selective about applying this -- specify exact cleaning cases for it rather than a catch-all.  In most cases, it probably means that something has gone wrong earlier in the area_id assignments.

Reply:
No -- in principle all clusters should be assigned to a subnational area.